### PR TITLE
[JENKINS-46760] - Disabled Agent Protocol monitor should not print warnings when TCP Agent port is disabled

### PR DESCRIPTION
**Pull request by oleg-nenashev**
See [JENKINS-46760](https://issues.jenkins-ci.org/browse/JENKINS-46760). I am not sure whether it's actually a bug, but I do not care much about instances without [Remoting] agents. So be it. Also added some annotations.

### Proposed changelog entries

* Entry 1: Disabled Agent Protocol monitor now does not print warnings when the TCP Agent Listener Port is disabled on the instance
* ...

### Submitter checklist

- [x] JIRA issue is well described
- [x] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@daniel-beck @reviewbybees

### CODE REVIEW

**Comment by daniel-beck** [comment]
Perhaps @jglick knows of a situation in which agent protocols without agent port are relevant?

**Comment by oleg-nenashev** [comment]
> Perhaps @jglick knows of a situation in which agent protocols without agent port are relevant?

We may need it when/if we have non-Tcp Agent Protocols. But it is a kind of moonshot for now.

**Comment by oleg-nenashev** [comment]
in order to support non-Tcp protocol cases I could add "requiresTcpAgentListener" to the protocol descriptor, but it is probably overkill for now

**Comment by jglick** [comment]
[Test failure](https://ci.jenkins.io/job/Core/job/jenkins/job/PR-3012/1/testReport/jenkins.slaves/DeprecatedAgentProtocolMonitorTest/noAdministrativeMonitorWhenPortIsDisabled/):

```
java.lang.AssertionError: Protocol(s) are not enabled: JNLP-connect,CLI-connect. Reason: Legacy Non-encrypted JNLP/CLI protocols should be enabled. Enabled protocols: CLI2-connect,JNLP2-connect,JNLP3-connect,JNLP4-connect,Ping
	at org.junit.Assert.fail(Assert.java:88)
	at jenkins.AgentProtocolTest.assertProtocols(AgentProtocolTest.java:127)
	at jenkins.slaves.DeprecatedAgentProtocolMonitorTest.noAdministrativeMonitorWhenPortIsDisabled(DeprecatedAgentProtocolMonitorTest.java:47)
```

**Review by jglick — COMMENTED** [review]
Other than test failure, sounds right. Of course it makes no sense to warn about protocols if we have turned off the service that would interpret those protocols.

**Comment by jglick** [comment]
> when/if we have non-Tcp Agent Protocols

That would be using an unrelated API. (Most likely we would never want such a thing because new code should be using HTTP.)

**Comment by oleg-nenashev** [comment]
> Most likely we would never want such a thing because new code should be using HTTP

I am not 100% sure about that, but it is a separate story. Even in such case you may need to show the new protocol and to be able to enable/disable it. And such logic should probably be a part of the existing UI. 

Technically it is possible to implement non-TCP protocol even now, so I would assume that such implementations may exist in non-public implementations. So I propose to say that the current behavior is valid and to invert the check in the added test.

**Comment by jglick** [comment]
> it is possible to implement non-TCP protocol even now

Sorry, how exactly?

> the current behavior is valid and to invert the check in the added test

Not following.

**Comment by oleg-nenashev** [comment]
Linux build passed, windows one prints something bizarre. Retriggering

```
<1s

Cloning the remote Git repository

Cloning with configured refspecs honoured and without tags

Loading of translation bundle failed for [org.eclipse.jgit.internal.JGitText, en_US]
```

**Comment by oleg-nenashev** [comment]
> Sorry, how exactly?

I wish I could show you an example. I am working with one of my previous employers to get it open-sourced.

**Comment by oleg-nenashev** [comment]
Anyway, my example was created before the rework of protocol stack by @stephenc. So probably it's not really applicable now. Since there is no real use-case in jenkinsci org right now, probably we should just merge it

**Review by jglick — DISMISSED** [review]
I filed [INFRA-1339](https://issues.jenkins-ci.org/browse/INFRA-1339) for the Windows CI error, but anyway this is still flaking:

```
[WARNING] jenkins.model.StartupTest.noWarnings(jenkins.model.StartupTest)
[ERROR]   Run 1: StartupTest.noWarnings:45 expected:<[]> but was:<[This Jenkins instance uses deprecated Remoting protocols: CLI-connect,CLI2-connect,JNLP-connect,JNLP2-connectIt may impact stability of the instance. If newer protocol versions are supported by all system components (agents, CLI and other clients), it is highly recommended to disable the deprecated protocols.]>
[INFO]   Run 2: PASS
```

I still maintain that `initializerCheck` should simply be deleted.

**Comment by jglick** [comment]
I have also seen `SetupWizardTest` flakes but I am not sure if they are related.

**Comment by oleg-nenashev** [comment]
OK, deleted it at all

**Comment by jglick** [comment]
Still failing:

```
java.lang.AssertionError: Protocol(s) are not enabled: JNLP-connect,CLI-connect. Reason: Legacy Non-encrypted JNLP/CLI protocols should be enabled. Enabled protocols: CLI2-connect,JNLP2-connect,JNLP3-connect,JNLP4-connect,Ping
	at org.junit.Assert.fail(Assert.java:88)
	at jenkins.AgentProtocolTest.assertProtocols(AgentProtocolTest.java:127)
	at jenkins.AgentProtocolTest.assertProtocols(AgentProtocolTest.java:105)
	at jenkins.AgentProtocolTest.testShouldNotDisableProtocolsForMigratedInstances(AgentProtocolTest.java:69)
```

File: core/src/main/java/jenkins/slaves/DeprecatedAgentProtocolMonitor.java [hunk 1]
```diff
@@ -102,17 +102,4 @@ public static String getDeprecatedProtocolsString() {
         }
         return StringUtils.join(deprecatedProtocols, ',');
     }
-    
-    @Initializer(after = InitMilestone.JOB_LOADED)
```
Comment (jglick) [inline core/src/main/java/jenkins/slaves/DeprecatedAgentProtocolMonitor.java]: BTW fix imports.

**Comment by jglick** [comment]
Also maybe time to fix `JenkinsTest`: https://github.com/jenkinsci/jenkins/pull/2621#pullrequestreview-65643816

**Comment by jglick** [comment]
```
java.lang.AssertionError: Deprecated Agent Protocol Monitor should not be activated
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertFalse(Assert.java:64)
	at jenkins.AgentProtocolTest.assertMonitorNotActive(AgentProtocolTest.java:133)
	at jenkins.install.SetupWizardTest.shouldDisableUnencryptedProtocolsByDefault(SetupWizardTest.java:126)
```

**Comment by oleg-nenashev** [comment]
@jglick Fixed the assumptions in #3118

File: core/src/main/java/jenkins/slaves/DeprecatedAgentProtocolMonitor.java [hunk 2]
```diff
@@ -97,17 +102,4 @@ public static String getDeprecatedProtocolsString() {
...
-    @Restricted(NoExternalUse.class)
-    public static void initializerCheck() {
```
Comment (jglick) [inline core/src/main/java/jenkins/slaves/DeprecatedAgentProtocolMonitor.java]: #3190

**Comment by oleg-nenashev** [comment]
I do not longer work on it

**Pull request closed without merging**